client/state/data-layer/wpcom-http/utils.js

```diff
@@ -42,11 +43,23 @@ export const getHeaders = action => get( action, 'meta.dataLayer.headers', null
  */
 export const getProgress = action => get( action, 'meta.dataLayer.progress', null );
 
+export const makeParser = ( schema, schemaOptions = {}, transformer = identity ) => {
+	const options = Object.assign( { verbose: true }, schemaOptions );
+	const validator = schemaValidator( schema, options );
+	const filter = schemaValidator.filter( schema );
+
+	return responseData => validator( responseData )
+		? [ true, transformer( filter( responseData ) ) ]
```

Comment:
can do that, though if it validated then I can't imagine the filter failing since it's built from the very validator itself.

the transformer could throw but it shouldn't. I would ascribe it to be similar to reducers where we also let exceptions bubble.

at this point I don't think we're in any promises